### PR TITLE
Add support for Juno

### PR DESCRIPTION
**Pull request by jenswi-linaro**
Signed-off-by: Jens Wiklander <jens.wiklander@linaro.org>

### CODE REVIEW

File: juno.mk [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+BUSYBOX_COMMON_CCDIR = $(AARCH64_PATH)
+
+busybox: busybox-common
+
+busybox-clean: busybox-clean-common
+
+busybox-cleaner: busybox-cleaner-common
+
+################################################################################
+# Das U-Boot
```
Comment (jbech-linaro) [inline juno.mk]: Das U-Boot :)

File: juno.mk [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+################################################################################
+
+U-BOOT_EXPORTS ?= CROSS_COMPILE="$(CCACHE)$(AARCH64_CROSS_COMPILE)"
+
+U-BOOT_DEFCONFIG_FILES := \
+	$(U-BOOT_PATH)/configs/vexpress_aemv8a_juno_defconfig \
+	$(ROOT)/kconfigs/u-boot_juno.conf
```
Comment (jbech-linaro) [inline juno.mk]: Is this line correct? Other local kconfig files (at least for kernel) are stored in build/kconfig/ ... which seems to be where you have put it when looking at the files further down.

File: juno.mk [hunk 3]
```diff
@@ -0,0 +1,192 @@
+DEBUG ?= 0
+
+-include common.mk
+
+################################################################################
+# Mandatory definition to use common.mk
+################################################################################
+CROSS_COMPILE_NS_USER	?= "$(CCACHE)$(AARCH64_CROSS_COMPILE)"
+CROSS_COMPILE_NS_KERNEL	?= "$(CCACHE)$(AARCH64_CROSS_COMPILE)"
```
Comment (pascal-brand-st-dev) [inline juno.mk]: strange space after the "

**Comment by pascal-brand-st-dev** [comment]
Reviewed-by: Pascal Brand <pascal.brand@linaro.org>

File: juno.mk (hunk 3 above)
Comment (jenswi-linaro) [inline juno.mk]: I'm sorry, I can find it.

**Comment by jbech-linaro** [comment]
@cedric-chaumont-st-dev would you mind testing this patch and the related PR in manifest.git? Just so we have another person trying it out also?

File: juno.mk (hunk 2 above)
Comment (jenswi-linaro) [inline juno.mk]: Thanks, I'll fix.

**Comment by jenswi-linaro** [comment]
The flash of the Juno board may need to be upgraded before trying this, check https://community.arm.com/docs/DOC-10804 (section 5.1) for instructions on how to upgrade using prebuilt binaries

File: juno.mk (hunk 3 above)
Comment (jbech-linaro) [inline juno.mk]: It's GitHub tricking Pascal here, it looks like there is a space (tab) between the `"` and `$(CCACHE)"`, but in fact there aren't if you're look at the raw patch.

**Comment by jbech-linaro** [comment]
With or without @cedric-chaumont-st doing a test:
`Reviewed-by: Joakim Bech <joakim.bech@linaro.org>`

**Comment by cedric-chaumont-st-dev** [comment]
Will test tomorrow.

**Comment by pascal-brand-st-dev** [comment]
@jenswi-linaro please squash and rebase

**Comment by jenswi-linaro** [comment]
Tag applied

**Pull request merged**